src/transfers.rs

```diff
@@ -87,34 +182,37 @@ fn make_transfers_map(transfers: Vec<Transfer>, sp: &CollectionWithId<StopPoint>
 
 fn generate_transfers_from_sp(
     transfers_map: &mut TransferMap,
-    stop_points: &CollectionWithId<StopPoint>,
+    model: &Model,
     max_distance: f64,
     walking_speed: f64,
     waiting_time: u32,
+    transfers_mode: &TransfersMode,
 ) {
     info!("Adding missing transfers from stop points.");
     let sq_max_distance = max_distance * max_distance;
-    for (idx1, sp1) in stop_points {
+    for (idx1, sp1) in model.stop_points.iter() {
         let approx = sp1.coord.approx();
-        for (idx2, sp2) in stop_points.iter() {
+        for (idx2, sp2) in model.stop_points.iter() {
             if transfers_map.contains_key(&(idx1, idx2)) {
                 continue;
             }
-            let sq_distance = approx.sq_distance_to(&sp2.coord);
-            if sq_distance > sq_max_distance {
-                continue;
+            if stop_points_need_transfer(model, idx1, idx2, transfers_mode, None) {
```

Comment:
what's the performance impact of this?

A continue would be more readable:

```rust
if stop_points_need_transfer(model, idx1, idx2, transfers_mode, None) {
    continue;
}
```

Reply:
the condition is stop_points_need_transfer(model, idx1, idx2, transfers_mode, None) AND sq_distance > sq_max_distance to continue

Reply:
regarding performances, we have for same data set transfers generation:
- on master 4.2s with 120M ram
- on this PR 6.9s with 124M ram

Reply:
On ffr-idf? Great!

Reply:
```rust
if !stop_points_need_transfer(model, idx1, idx2, transfers_mode, None) {
    continue;
}
if sq_distance > sq_max_distance {
    continue;
}
```

Reply:
ah ok !